### ISSUE TITLE
Avionics improvments

### ISSUE THREAD

**Issue by Zarbizaure** [KSP-RO/RP-0#560]
So many times I lost a vessel because it was weighting only a few kilograms over the avionic limit. This happens really often: I set my fuel to be just under the mass limit (example: 1T for the gold Hecs core), then add a tiny instrument...and I'm over the mass limit, which kills the whole mission. That's why I had an idea to solve that.

Instead of completely disabling controls, overweighting the max avionics mass should just add a virtual signal lag (like RT does). Indeed, a computer overwhelmed by informations, complex engines data, telemetry data and sensors (what will happen if you let the tiny early probe core control a saturn V rocket) will still be able to compute, but it will be much slower (eventually it'll crash because it lacks RAM/storage, but that's the worst case).

Speaking of in-game implementation, the signal delay (if vesselMass > maxAvionicsMass) should be something like:

```
t = ((vesselMass - maxAvionicsMass)/maxAvionicsMass)^n
```

with a n a number that we'll choose to have an adapted curve, I think a square or cubic function (n = 3) would be cool.

That way, if 

```
vesselMass = 2 x maxAvionicsMass then signal delay = RTdelay +1s
vesselMass = 3 x maxAvionicsMass then signal delay = RTdelay +8s
vesselMass = 4 x maxAvionicsMass then signal delay = RTdelay +27s
...
```

Anyway, that's just an idea to make the avionic system a bit more tolerant when you're only a few kilograms over your mass limit.

Other that that, we really need some advanced avionics in the 2m and 3m versions, and eventually procedural avionics (but I heard that people are already working on this).

**Comment by infinitybit** [comment]
In my opinion the concept of avionics as currently implemented if fundamentally flawed. It might be due to my near total ignorance on the subject but I don't see how mass is a limiting factor. It seems like it was an arbitrary choice to fit gameplay mechanics.  If so then it should either be changed to a better system or simply scrapped.
I haven't tried it but is there anything keeping me from putting 1950's vintage guidance unit into Falcon 9?  Would it be realistic?
In my understanding, the primary limitation in the early days was computing power and sensors.  Rockets were fast and computers were slow. Modern guidance system has access to things like precision clocks, IMUs, start tracking, etc. Today an Arduino could probably guide Saturn V into orbit.
Part of this avionics mess is tied to gameplay mechanic of flying the rocket by hand.  Im pretty sure SpaceX doesn't have a dude sitting at a stick and staring at a NavBall.
Real life guidance systems are either fully autonomous, executing pre-programmed sequence or semi-autonomous, receiving commands from ground station.  In-game avionics as implemented don't really do their job of guiding rocket to orbit.  Instead they simply impose arbitrary mass limit on manual control.
TLDR
I have a couple of ideas for better solution but I doubt they are feasible, so i won't list them here.
Instead, I would just rip out the mass limit and simply require "level x" avionics unit.  If you fly with Tier 6 hardware then you need Tier 6 avionics, and that is that.  Call it compatibility.

**Comment by pap1723** [comment]
Anyone have any thoughts on this, or different ways avionics could be used?

**Comment by pap1723** [comment]
This is still a clever approach to avionics IMO.

**Comment by MikeOnTea** [comment]
> This is still a clever approach to avionics IMO.
Which idea are you talking about? In any case at first thought i don't like either idea too much.
Another idea: Use engine types and total thrust instead of mass. Would allow you to do stuff like require avionics to be tooled for certain engines and it would make it harder to accidentally get over the limit, but overall probably quite a lot of work for little gain.

**Comment by pap1723** [comment]
Yeah, probably all true.
